### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by ViachaslauKabak**
Refactor the code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,8 @@ Solr Configuration as Master
 =========
 [![License](https://img.shields.io/badge/license-Apache-green.svg?style=flat)](https://raw.githubusercontent.com/lean-delivery/ansible-role-solr-master/master/LICENSE)
 [![Build Status](https://travis-ci.org/lean-delivery/ansible-role-solr-master.svg?branch=master)](https://travis-ci.org/lean-delivery/ansible-role-solr-master)
-[![Galaxy](https://img.shields.io/badge/galaxy-lean__delivery.solr__master-blue.svg)](https://galaxy.ansible.com/lean_delivery/solr_master)
+[![Build Status](https://gitlab.com/lean-delivery/ansible-role-solr-master/badges/master/build.svg)](https://gitlab.com/lean-delivery/ansible-role-solr-master)
```
Comment (tgadiev) [inline README.md]:
```suggestion
[![Build Status](https://gitlab.com/lean-delivery/ansible-role-solr-master/badges/master/build.svg)](https://gitlab.com/lean-delivery/ansible-role-solr-master/pipelines)
```

File: README.md [hunk 2]
```diff
@@ -32,29 +33,28 @@ Requirements
...
-  - `override_dest_main_path` - root directory to store solr folder
-    default: `/opt`
-    default: `C:\Solr`
-  - `override_dest_solr_path` - solr folder path
-    default: `{{ dest_main_path }}/solr-{{ solr_version }}`
-    default: `{{ dest_main_path }}\\solr-{{ solr_version }}`
-  - `overrride_solr_configset_path` - solr configset folder path
-    default: `{{ solr_home }}/configsets`
-    default: `{{ dest_solr_path }}\\server\\solr\\configsets`
+  - `solr_version` - matches available version on https://archive.apache.org/dist/lucene/solr/. Tested versions 5.3-8.x
```
Comment (tgadiev) [inline README.md:36]:
```suggestion
  - `solr_version` - matches available version on https://archive.apache.org/dist/lucene/solr/. Tested versions 5.3-8.x

```

File: README.md [hunk 3]
```diff
@@ -32,29 +33,28 @@ Requirements
...
+    default: `8.0.0`
+  - `solr_dest_main_path` - root directory to store solr folder
```
Comment (tgadiev) [inline README.md:40]:
```suggestion
  - `solr_dest_main_path` - root directory to store solr folder

```

File: README.md [hunk 4]
```diff
@@ -32,29 +33,28 @@ Requirements
...
+    default: `/opt` for Linux
```
Comment (tgadiev) [inline README.md]:
```suggestion
    default: `/opt` for Linux

```

File: README.md [hunk 5]
```diff
@@ -32,29 +33,28 @@ Requirements
...
-    default: `{{ dest_main_path }}\\solr-{{ solr_version }}`
-  - `overrride_solr_configset_path` - solr configset folder path
-    default: `{{ solr_home }}/configsets`
-    default: `{{ dest_solr_path }}\\server\\solr\\configsets`
+  - `solr_version` - matches available version on https://archive.apache.org/dist/lucene/solr/. Tested versions 5.3-8.x
+    default: `8.0.0`
+  - `solr_dest_main_path` - root directory to store solr folder
+    default: `/opt` for Linux  
+    default: `C:\Solr` for Windows  
+  - `solr_dest_path` - solr folder path
```
Comment (tgadiev) [inline README.md:46]:
```suggestion
  - `solr_dest_path` - solr folder path

```

File: README.md [hunk 6]
```diff
@@ -32,29 +33,28 @@ Requirements
...
+    default: `{{ dest_main_path }}/solr-{{ solr_version }}`  
+  - `solr_configset_path` - solr configset folder path
```
Comment (tgadiev) [inline README.md:50]:
```suggestion
  - `solr_configset_path` - solr configset folder path

```

File: README.md [hunk 7]
```diff
@@ -32,29 +33,28 @@ Requirements
...
+    default: `{{ solr_home }}/configsets` for Linux
```
Comment (tgadiev) [inline README.md]:
```suggestion
    default: `{{ solr_home }}/configsets` for Linux   
```

File: README.md [hunk 8]
```diff
@@ -32,29 +33,28 @@ Requirements
...
+    default: `8.0.0`
+  - `solr_dest_main_path` - root directory to store solr folder
+    default: `/opt` for Linux  
+    default: `C:\Solr` for Windows  
+  - `solr_dest_path` - solr folder path
+    default: `{{ dest_main_path }}/solr-{{ solr_version }}`  
+  - `solr_configset_path` - solr configset folder path
+    default: `{{ solr_home }}/configsets` for Linux  
+    default: `{{ solr_dest_path }}\server\solr\configsets` for Windows  
   - `solr_service_name` - solr service name
```
Comment (tgadiev) [inline README.md:56]:
```suggestion
  - `solr_service_name` - solr service name

```

File: README.md [hunk 9]
```diff
@@ -32,29 +33,28 @@ Requirements
...
-    default: `solr`
+    default: `solr`  
   - `solr_with_systemd` - to run solr as a service
```
Comment (tgadiev) [inline README.md:60]:
```suggestion
  - `solr_with_systemd` - to run solr as a service

```

File: README.md [hunk 10]
```diff
@@ -32,29 +33,28 @@ Requirements
...
-    default: `True`
+    default: `True`  
   - `configset_list` - list of configset directories
```
Comment (tgadiev) [inline README.md:64]:
```suggestion
  - `configset_list` - list of configset directories

```

File: README.md [hunk 11]
```diff
@@ -32,29 +33,28 @@ Requirements
...
-    default: `- default`
+    default: `- default`  
   - `auto_populate_configset_list` - get all configset directories automatically
```
Comment (tgadiev) [inline README.md:68]:
```suggestion
  - `auto_populate_configset_list` - get all configset directories automatically

```

File: README.md [hunk 12]
```diff
@@ -32,29 +33,28 @@ Requirements
...
-    default: `True`
+    default: `True`  
   - `solr_base_path` - path to solr base
```
Comment (tgadiev) [inline README.md:72]:
```suggestion
  - `solr_base_path` - path to solr base

```

File: README.md [hunk 13]
```diff
@@ -32,29 +33,28 @@ Requirements
...
-    default: `/var/solr`
+    default: `/var/solr`  
   - `solr_home` - path to SOLR_HOME
```
Comment (tgadiev) [inline README.md:76]:
```suggestion
  - `solr_home` - path to SOLR_HOME

```

File: handlers/main.yml [hunk 14]
```diff
@@ -1,12 +1,12 @@
 ---
-- name: "restart solr linux"
+- name: restart solr linux
```
Comment (tgadiev) [inline handlers/main.yml]:
```suggestion
- name: restart Solr Linux
```

File: handlers/main.yml [hunk 15]
```diff
@@ -1,12 +1,12 @@
...
   service:
-    name: "{{ solr_service_name }}"
+    name: '{{ solr_service_name }}'
     state: restarted
   when: solr_with_systemd
   become: True
 
-- name: "restart solr windows"
+- name: restart solr windows
```
Comment (tgadiev) [inline handlers/main.yml]:
```suggestion
- name: restart Solr Windows
```

File: meta/main.yml [hunk 16]
```diff
@@ -5,13 +5,12 @@ galaxy_info:
   description: "Lean Delivery Solr configuration as master"
   company: "Epam Systems"
   license: "Apache"
-  min_ansible_version: "2.5"
+  min_ansible_version: "2.7"
   issue_tracker_url: "https://github.com/lean-delivery/ansible-role-solr-master/issues"
   platforms:
     - name: "Ubuntu"
       versions:
         - "xenial"
```
Comment (tgadiev) [inline meta/main.yml:13]: Only "xenial"? Why not "bionic"?

File: molecule/cloud-aws-direct/molecule.yml [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+      solr8:
+        solr_version: 8.0.0
+  playbooks:
+    create: ../resources/provisioning/AWS/create.yml
+    prepare: ../resources/provisioning/AWS/prepare.yml
+    destroy: ../resources/provisioning/AWS/destroy.yml
+  lint:
+    name: ansible-lint
+  env:
+    ANSIBLE_LIBRARY: ${ANSIBLE_LIBRARY}
```
Comment (tgadiev) [inline molecule/cloud-aws-direct/molecule.yml]: Not necessary for AWS anymore

File: molecule/cloud-aws-windows/molecule.yml [hunk 18]
```diff
@@ -0,0 +1,71 @@
...
+driver:
+  name: delegated
+lint:
+  name: yamllint
+  options:
+    config-file: .yamllint
+platforms:
+
+  - name: test-aws-Windows2016-solr-master-solr7
+    aws_image_id: ami-07a29e78aeb420471
```
Comment (tgadiev) [inline molecule/cloud-aws-windows/molecule.yml]:
```suggestion
    aws_image_id: ami-00a20f508263efd30
```

File: molecule/cloud-aws-windows/molecule.yml [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+    spot_price: 0.14
+    aws_sec_group:
+      - default
+    wait_timeout: 1800
+    groups:
+      - windows
+      - solr7
+
+  - name: test-aws-Windows2016-solr-master-solr8
+    aws_image_id: ami-07a29e78aeb420471
```
Comment (tgadiev) [inline molecule/cloud-aws-windows/molecule.yml]:
```suggestion
    aws_image_id: ami-00a20f508263efd30
```

File: molecule/cloud-aws-windows/molecule.yml [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+        ansible_password: "{{ lookup('env', 'WINRM_ADMIN_PWD') | default('M0leCule123!', True) }}"
+        ansible_winrm_transport: credssp
+        ansible_winrm_server_cert_validation: ignore
+  playbooks:
+    create: ../resources/provisioning/WINRM/create.yml
+    destroy: ../resources/provisioning/WINRM/destroy.yml
+  lint:
+    name: ansible-lint
+  env:
+    ANSIBLE_LIBRARY: ${ANSIBLE_LIBRARY}
```
Comment (tgadiev) [inline molecule/cloud-aws-windows/molecule.yml]: Not necessary for AWS

**Review by tgadiev — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (ViachaslauKabak) [reply README.md:36]: fixed

File: README.md (hunk 3 above)
Comment (ViachaslauKabak) [reply README.md:40]: fixed

File: README.md (hunk 5 above)
Comment (ViachaslauKabak) [reply README.md:46]: fixed

File: meta/main.yml (hunk 16 above)
Comment (ViachaslauKabak) [reply meta/main.yml:13]: solr-standalone supports only xenial
we do not test on bionic

**Pull request merged**